components/match2/commons/match_group_input.lua

```diff
@@ -241,6 +241,38 @@ function MatchGroupInput.getInexactDate(suggestedDate)
 	return getContentLanguage():formatDate('c', inexactDateString)
 end
 
+--[[
+Parses the match group context. The match group context describes where a match
+group is relative to the tournament page.
+]]
+function MatchGroupInput.readContext(args)
+	local context = {
+		bracketIndex = tonumber(globalVars:get('match2bracketindex')) or 0,
+		groupRoundIndex = MatchGroupInput.readGroupRoundIndex(args),
+		matchSection = args.matchsection or globalVars:get('matchsection'),
+		sectionHeader = args.section or globalVars:get('bracket_header'),
+		tournamentParent = globalVars:get('tournament_parent'),
+	}
+
+	globalVars:set('bracket_header', context.sectionHeader)
+	globalVars:set('matchsection', context.matchSection)
+
+	return context
+end
+
+function MatchGroupInput.readGroupRoundIndex(args)
+	if args.round then
+		return tonumber(args.round)
+	end
+
+	local matchSection = args.matchsection or globalVars:get('matchsection')
+	-- Usually 'Round N' but can also be 'Day N', 'Week N', etc.
+	local roundIndex = matchSection and matchSection:match(' (%d+)$')
+	if roundIndex then
+		return tonumber(roundIndex)
+	end
```

Comment:
Return default value? Perhaps `-1`?

Reply:
I think it's better to store `nil` (i.e. no storage) if match section isn't set. That means this should return `nil` instead of `-1`. I'll change it to explicitly return `nil`.